### ISSUE TITLE
publish to gradle plugins portal

### ISSUE THREAD

**Issue by gotson** [societe-generale/arch-unit-gradle-plugin#9]
The plugin is not available on the [Gradle plugin portal](https://plugins.gradle.org/), preventing the use of the [plugins DSL](https://docs.gradle.org/current/userguide/plugins.html#sec:plugins_block).

Publishing [seems easy](https://plugins.gradle.org/docs/submit), but requires an API key.

**Comment by vincent-fuchs** [comment]
indeed, doesn't seem to complicated !

just did it in https://github.com/societe-generale/arch-unit-gradle-plugin/commit/b4055dac675bbd97be7ef2275f85f313d285580f

looks like it worked : 

_Publishing plugin com.societegenerale.common.arch-unit-gradle-plugin version 1.0.4-SNAPSHOT
Thank you. Your new plugin com.societegenerale.common.arch-unit-gradle-plugin has been submitted for approval by Gradle engineers. The request should be processed within the next 24 hours, at which point you will be contacted via email._

I'll update and close the PR once they get back to me

**Comment by vincent-fuchs** [comment]
OK, looks like it kind of worked.. "kind of", because I ran ```gradlew release publishPlugins -Prelease.useAutomaticVersion=true``` and it created a v1.0.4 (and then upgraded the version to 1.0.5-SNAPSHOT), but it's a 1.0.4-SNAPSHOT that got uploaded to plugin repository, see https://plugins.gradle.org/plugin/com.societegenerale.common.arch-unit-gradle-plugin 

Any idea how to avoid this and publish the version that I am releasing with the ```net.researchgate.release``` plugin ?

**Comment by gotson** [comment]
The Gradle [`publish-plugin`](https://plugins.gradle.org/docs/publish-plugin) will use the project version:

```
// Unless overridden in the pluginBundle config DSL, the project version will
// be used as your plugin version when publishing
```

I don't know about the `net.researchgate.release` plugin nor how it increments your release number.

Do you increment version number at every commit, or at specific times using a manual workflow (like a tag or something)?

it seems you also reran the `release` task, so it may have bumped your version to 1.0.5 before publishing to the plugin portal.

You could run only the `publishPlugins` task maybe.

**Comment by vincent-fuchs** [comment]
ok, fixed it ! v1.0.4 is available on https://plugins.gradle.org/plugin/com.societegenerale.common.arch-unit-gradle-plugin !

I added a task dependency : https://github.com/societe-generale/arch-unit-gradle-plugin/blob/c1af2ed56cff834dfbe7336e907029889e684401/build.gradle#L130
